Tests/ApolloCodegenTests/CodeGeneration/Templates/EnumTemplateTests.swift

```diff
@@ -14,9 +15,23 @@ class EnumTemplateTests: XCTestCase {
 
   // MARK: Helpers
 
-  private func buildSubject(name: String = "testEnum", values: [String] = ["ONE", "TWO"]) {
+  /// Convenience function to build the `EnumTemplate` `subject`.
+  ///
+  /// - Parameters:
+  ///   - name: Enum definition name.
+  ///   - values: A tuple that combines the value name and optional deprecation reason.
+  ///   - config: Code generation configuration.
+  private func buildSubject(
+    name: String = "testEnum",
+    values: [(String, String?)] = [("ONE", nil), ("TWO", nil)],
+    config: ApolloCodegenConfiguration = ApolloCodegenConfiguration.mock()
+  ) {
     subject = EnumTemplate(
-      graphqlEnum: GraphQLEnumType.mock(name: name, values: values)
+      graphqlEnum: GraphQLEnumType.mock(
+        name: name,
+        values: values.map({ GraphQLEnumValue.mock(name: $0.0, deprecationReason: $0.1) })
+      ),
+      config: ReferenceWrapped(value: config)
```

Comment:
We should really fix `ReferenceWrapped` to not have the `value` name on the initializer. I'm going to do that at some point.

Reply:
At the time I thought it read nicely and fit with the Swift naming guidelines. Not opposed to renaming it though.